defaults/cmake/Key4hepConfig.cmake

```
@@ -0,0 +1 @@
+set(CMAKE_CXX_FLAGS "-fPIC -Wall -Wextra -Wpedantic ${CMAKE_CXX_FLAGS}")
```

Check that each flag is supported by the compiler like https://github.com/AIDASoft/DD4hep/blob/0c668ca1d6716eb12b06cc8c90b34e04079ff014/cmake/DD4hepBuild.cmake#L39
```suggestion
set(CMAKE_CXX_FLAGS "-fPIC -Wall -Wextra -Wpedantic  -Wdeprecated -fdiagnostics-color=auto -Wshadow ${CMAKE_CXX_FLAGS}")
```

For clang `-Wheader-hygiene`

---

Added